src/app/shared/dia-backend/asset/prefetching/dia-backend-asset-prefetching.service.ts

```diff
@@ -8,12 +9,17 @@ import { DiaBackendAssetDownloadingService } from '../downloading/dia-backend-do
 export class DiaBackendAssetPrefetchingService {
   constructor(
     private readonly assetRepository: DiaBackendAssetRepository,
-    private readonly downloadingService: DiaBackendAssetDownloadingService
+    private readonly downloadingService: DiaBackendAssetDownloadingService,
+    private readonly proofRepository: ProofRepository
   ) {}
 
   async prefetch(
     onStored: (currentCount: number, totalCount: number) => any = () => void 0
   ) {
+    const proofs = await this.proofRepository.getAll();
+    const cachedAssetThumbnailCids = new Set();
+    proofs.map(proof => cachedAssetThumbnailCids.add(proof.diaBackendAssetId));
```

Comment:
I don't see any function that check if an asset exist in `ProofRepository` since our database implementation doesn't support lookup by index. This is why I want to migrate our db to RxDB or something else.